branchLayoutImpl/src/main/java/com/virtuslab/branchlayout/impl/BranchLayout.java

```diff
@@ -76,24 +102,30 @@ private BaseBranchLayoutEntry updateSubbranchesForEntry(BaseBranchLayoutEntry en
   }
 
   private Option<BaseBranchLayoutEntry> findUpstreamEntryForEntry(BaseBranchLayoutEntry entry) {
-    return findEntryRecursively(rootBranches, e -> e.getSubbranches().contains(entry));
+    return findEntryRecursively(rootBranches, e -> e.getSubentries().contains(entry));
   }
 
   /** Recursively traverses the list for an element that satisfies the {@code predicate}. */
   private static Option<BaseBranchLayoutEntry> findEntryRecursively(
-      List<BaseBranchLayoutEntry> branches,
+      List<BaseBranchLayoutEntry> entries,
       Predicate<BaseBranchLayoutEntry> predicate) {
-    for (var branch : branches) {
-      if (predicate.test(branch)) {
-        return Option.of(branch);
+    LOG.debug(() -> "Enter findEntryRecursively(entries = ${entries}, predicate = ${predicate})");
```

Comment:
There's likely little point in printing the `predicate`, it won't give any valuable info